src/solver/main/adequacy-draft.cpp

```diff
@@ -25,17 +25,24 @@
 ** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
 */
 
-#include "../main.h"
+#include "../application.h"
 #include "../simulation/solver.h"
 #include "../simulation/adequacy-draft.h"
 #include <antares/logs.h>
 
-using namespace Antares;
-
-void SolverApplication::runSimulationInAdequacyDraftMode()
+namespace Antares
+{
+namespace Solver
+{
+void Application::runSimulationInAdequacyDraftMode()
 {
     // Type of the simulation
     typedef Solver::Simulation::ISimulation<Solver::Simulation::AdequacyDraft> SimulationType;
-
-    runSimulation<SimulationType>();
+    SimulationType simulation(*pStudy, pSettings);
```

Comment:
There is code duplication here : we have same lines of code for draft, adequacy and economy

Reply:
Right, except that SimulationType is not the same. I removed the function template.

Reply:
Function templates require an **.hxx** file, adding some complexity and compilation time. I'd rather not use them if possible.

Reply:
Interface + factory function